diff --git a/README.md b/README.md
index 4c5a681..f68d290 100644
--- a/README.md
+++ b/README.md
@@ -124,3 +124,4 @@ Then execute the StartRun() command to start the triggering
 ```
 
 <!--- ######################################################## -->
+
diff --git a/firmware/python/high_rate_encoder_dev/__init__.py b/firmware/python/high_rate_encoder_dev/__init__.py
index 535b7f1..6e90d66 100644
--- a/firmware/python/high_rate_encoder_dev/__init__.py
+++ b/firmware/python/high_rate_encoder_dev/__init__.py
@@ -1,5 +1,9 @@
 #!/usr/bin/env python
+import sys
+import os
+sys.path.append(os.path.dirname(os.path.realpath(__file__)))
 from high_rate_encoder_dev._Fmc  import *
 from high_rate_encoder_dev._App  import *
 from high_rate_encoder_dev._Core import *
 from high_rate_encoder_dev._Root import *
+
diff --git a/setup.py b/setup.py
index 7d0c7ab..f2958b3 100644
--- a/setup.py
+++ b/setup.py
@@ -11,18 +11,15 @@ setup(
     description = 'High Rate econder  package',
     packages = [
         'high_rate_encoder_dev',
-        'high_rate_encoder_dev.axipcie',
+        'high_rate_encoder_dev.LclsTimingCore',
         'high_rate_encoder_dev.l2si_core',
-        'high_rate_encoder_dev.daq_stream_cache',
-        'high_rate_encoder_dev.lcls2_pgp_fw_lib.hardware.XilinxKcu1500',
-        'high_rate_encoder_dev.lcls2_pgp_fw_lib.hardware.SlacPgpCardG4',
-        'high_rate_encoder_dev.lcls2_pgp_fw_lib.hardware',
-        'high_rate_encoder_dev.lcls2_pgp_fw_lib.hardware.shared',
-        'high_rate_encoder_dev.lcls2_pgp_fw_lib',
+        'high_rate_encoder_dev.surf',
         'high_rate_encoder_dev.surf.misc',
+        'high_rate_encoder_dev.axipcie',
+        'high_rate_encoder_dev.lcls2_pgp_fw_lib',
+        'high_rate_encoder_dev.lcls2_pgp_fw_lib.shared',
         'high_rate_encoder_dev.surf.xilinx',
         'high_rate_encoder_dev.surf.axi',
-        'high_rate_encoder_dev.surf',
         'high_rate_encoder_dev.surf.ethernet.xaui',
         'high_rate_encoder_dev.surf.ethernet.udp',
         'high_rate_encoder_dev.surf.ethernet.ten_gig',
@@ -40,26 +37,30 @@ setup(
         'high_rate_encoder_dev.surf.protocols.batcher',
         'high_rate_encoder_dev.surf.protocols.rssi',
         'high_rate_encoder_dev.surf.protocols.pgp',
+        'high_rate_encoder_dev.surf.protocols.htsp',
+        'high_rate_encoder_dev.surf.protocols.sugoi',
+        'high_rate_encoder_dev.surf.protocols.coaxpress',
+        'high_rate_encoder_dev.surf.protocols.ssp',
         'high_rate_encoder_dev.surf.devices.intel',
         'high_rate_encoder_dev.surf.devices.nxp',
         'high_rate_encoder_dev.surf.devices.cypress',
         'high_rate_encoder_dev.surf.devices.linear',
         'high_rate_encoder_dev.surf.devices.micron',
+        'high_rate_encoder_dev.surf.devices.maxim',
         'high_rate_encoder_dev.surf.devices',
         'high_rate_encoder_dev.surf.devices.analog_devices',
         'high_rate_encoder_dev.surf.devices.microchip',
         'high_rate_encoder_dev.surf.devices.transceivers',
         'high_rate_encoder_dev.surf.devices.silabs',
         'high_rate_encoder_dev.surf.devices.ti',
-        'high_rate_encoder_dev.LclsTimingCore',
+        'high_rate_encoder_dev.surf.devices.amphenol',
     ],
     package_dir = {
+        'high_rate_encoder_dev.l2si_core': 'firmware/submodules/l2si-core/python/l2si_core',
+        'high_rate_encoder_dev.LclsTimingCore': 'firmware/submodules/lcls-timing-core/python/LclsTimingCore',
         'high_rate_encoder_dev': 'firmware/python/high_rate_encoder_dev',
         'high_rate_encoder_dev.surf': 'firmware/submodules/surf/python/surf',
-        'high_rate_encoder_dev.surf': 'firmware/submodules/surf/python/surf',
         'high_rate_encoder_dev.axipcie': 'firmware/submodules/axi-pcie-core/python/axipcie',
-        'high_rate_encoder_dev.LclsTimingCore': 'firmware/submodules/lcls-timing-core/python/LclsTimingCore',
         'high_rate_encoder_dev.lcls2_pgp_fw_lib': 'firmware/submodules/lcls2-pgp-fw-lib/python/lcls2_pgp_fw_lib',
-        'high_rate_encoder_dev.l2si_core': 'firmware/submodules/l2si-core/python/l2si_core'
     }
 )
